### PR TITLE
Ci integration

### PR DESCRIPTION
**Pull request by Olthoff231381**
Many commits to get the CI working reliably but it now works

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Getting repository
+      uses: actions/checkout@v4
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
```
Comment (iulusoy) [inline .github/workflows/ci.yml:20]: You generally don't need this, if you pip install the package below, that should give you all the requirements.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    - name: Install pytest and package
+      run: |
+        pip install pytest
+        pip install -e .
+    - name: Execute PyTest
+      run: pytest
```
Comment (iulusoy) [inline .github/workflows/ci.yml:26]: To be on the safe side, I would run `python -m pytest` and you can include options like `-svv` to make any failures more apparent and understandable.

**Review by iulusoy — CHANGES_REQUESTED** [review]
Two small items, looks great otherwise.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Olthoff231381) [reply .github/workflows/ci.yml:20]: So just leave out the 'pip install -r ...' ?

Comment (iulusoy) [reply .github/workflows/ci.yml:20]: Only the `pip install -r requirements.txt`. Instead, you can directly put `pip install -e .`

**Review by Olthoff231381 — COMMENTED** [review]
Deleted requirements install and added flags to pytest

**Comment by Olthoff231381** [comment]
Will fix the Quality Problem next push.
Just have to omit the conversion to path in parse.py

**Pull request merged**